src/lib/containers/entity/annotations/AdditionalPropertiesSchemaField.tsx

```diff
@@ -49,10 +52,21 @@ export function transformDataFromPropertyType(
 ) {
   switch (propertyType) {
     case PropertyType.INTEGER:
-    case PropertyType.FLOAT:
       return list.map(item =>
-        Number.isNaN(Number(item)) ? undefined : Number(item),
+        Number.isNaN(Number(item)) ? undefined : Math.floor(Number(item)),
       )
+
+    case PropertyType.FLOAT:
+      return list.map(item => {
+        const asFloat = parseFloat(item)
+        if (Number.isNaN(asFloat)) {
+          return 'NaN'
+        } else if (Number.isInteger(asFloat)) {
+          return `${asFloat}.0`
```

Comment:
The backend drops precision, so float values may become integers. We add the '.0' so the backend will re-accept it as a float.

Reply:
This is SWC-5758